### PR TITLE
[bitnami/external-dns] Support common labels for all resources

### PR DESCRIPTION
**Pull request by thib-mary**
Usefull when we have a policies manager as Kyverno.

Signed-off-by: Flaagada <mary.thibault2@gmail.com>

### Description of the change

Support a common labels for all resources.

### Benefits

When we used a policies manager as Kyverno, it's usefull to be able to add a common labels on all resources created by the chart in order to respect the policy.

### Checklist

- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/external-dns/values.yaml [hunk 1]
```diff
@@ -14,6 +15,10 @@ global:
   ##   - myRegistryKeySecretName
   ##
   imagePullSecrets: []
+    
+  commonLabels: {}
+  ## e.g:
+  ## team_name: dev
```
Comment (carrodher) [inline bitnami/external-dns/values.yaml:21]: In order to follow the same approach used by the rest of parameters:
```suggestion
  ## E.g.
  ## commonLabels:
  ##   team_name: dev
  ##
  commonLabels: {}
```

**Review by carrodher — COMMENTED** [review]
Thanks for your contribution, I just added a minor suggestion

File: bitnami/external-dns/values.yaml [hunk 2]
```diff
@@ -14,6 +15,12 @@ global:
   ##   - myRegistryKeySecretName
   ##
   imagePullSecrets: []
+    
+  commonLabels: {}
+  ## E.g.
+  ## commonLabels:
+  ##   team_name: dev
+  ##
```
Comment (carrodher) [inline bitnami/external-dns/values.yaml:23]: Could you please also take a look at the position of the example? (You can simply apply this suggestion so you don't to do any change)
```suggestion
  imagePullSecrets: []
  ## E.g.
  ## commonLabels:
  ##   team_name: dev
  ##
  commonLabels: {}
```

**Comment by Flaagada** [comment]
> On the other hand, please note there is already a `commonLabels` parameters available
> 
> https://github.com/bitnami/charts/blob/2584f4a12db4848ecb41db76023149f818ce9762/bitnami/external-dns/values.yaml#L31

In fact, it doesn't work on [secrets](https://github.com/bitnami/charts/blob/2584f4a12db4848ecb41db76023149f818ce9762/bitnami/external-dns/templates/secret.yaml) and [configmap](https://github.com/bitnami/charts/blob/2584f4a12db4848ecb41db76023149f818ce9762/bitnami/external-dns/templates/configmap.yaml)

**Comment by carrodher** [comment]
Well, that is a different topic then. We should make the current parameter work in all cases rather than creating a new one for the same purpose.

In the case of secrets, the custom labels are added without any issue.
I used the following values.yaml
```yaml
commonLabels:
  custom: custom
provider: "google"
google:
  serviceAccountKey: "test"
```
and rendered the templates by running
```console
$ helm template bitnami/external-dns --generate-name -f my-values.yaml
...
# Source: external-dns/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: release-name-external-dns
  namespace: "default"
  labels:
    app.kubernetes.io/name: external-dns
    helm.sh/chart: external-dns-6.5.5
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    custom: custom
  annotations:
type: Opaque
data:
  credentials.json: "dGVzdA=="
```
As you can see, the `custom: custom` label is taken into account without any problem. But please, note secrets are not rendered out of the box, you need to specify the provider and associated parameters

**Comment by thib-mary** [comment]
> Well, that is a different topic then. We should make the current parameter work in all cases rather than creating a new one for the same purpose.
> 
> In the case of secrets, the custom labels are added without any issue. I used the following values.yaml
> 
> ```yaml
> commonLabels:
>   custom: custom
> provider: "google"
> google:
>   serviceAccountKey: "test"
> ```
> 
> and rendered the templates by running
> 
> ```
> $ helm template bitnami/external-dns --generate-name -f my-values.yaml
> ...
> # Source: external-dns/templates/secret.yaml
> apiVersion: v1
> kind: Secret
> metadata:
>   name: release-name-external-dns
>   namespace: "default"
>   labels:
>     app.kubernetes.io/name: external-dns
>     helm.sh/chart: external-dns-6.5.5
>     app.kubernetes.io/instance: release-name
>     app.kubernetes.io/managed-by: Helm
>     custom: custom
>   annotations:
> type: Opaque
> data:
>   credentials.json: "dGVzdA=="
> ```
> 
> As you can see, the `custom: custom` label is taken into account without any problem. But please, note secrets are not rendered out of the box, you need to specify the provider and associated parameters. Something similar happens with configmaps, it is not rendered by default unless you set the proper parameters to deploy a configmap

Ok, my bad, i close the PR 👍 . Thanks!

**Pull request closed without merging**